subsys/logging/log_backend_spinel.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log_backend.h>
+#include <logging/log_output.h>
+#include <openthread/platform/logging.h>
+#include "log_backend_std.h"
+
+#ifndef CONFIG_LOG_BACKEND_SPINEL_BUFFER_SIZE
+#define CONFIG_LOG_BACKEND_SPINEL_BUFFER_SIZE 0
+#endif
+
+static uint8_t char_buf[CONFIG_LOG_BACKEND_SPINEL_BUFFER_SIZE];
+
+static int write(uint8_t *data, size_t length, void *ctx);
+
+LOG_OUTPUT_DEFINE(log_output, write,
+		  char_buf, sizeof(char_buf));
+
+static void put(const struct log_backend *const backend,
+		struct log_msg *msg)
+{
+	/* prevent adding CRLF, which may crash spinel decoding */
+	uint32_t flag = LOG_OUTPUT_FLAG_CRLF_NONE;
+
+	log_backend_std_put(&log_output, flag, msg);
+}
+
+static void sync_string(const struct log_backend *const backend,
+		     struct log_msg_ids src_level, uint32_t timestamp,
+		     const char *fmt, va_list ap)
+{
+	/* prevent adding CRLF, which may crash spinel decoding */
+	uint32_t flag = LOG_OUTPUT_FLAG_CRLF_NONE;
+
+	log_backend_std_sync_string(&log_output, flag, src_level,
+				    timestamp, fmt, ap);
+}
+
+static void sync_hexdump(const struct log_backend *const backend,
+			 struct log_msg_ids src_level, uint32_t timestamp,
+			 const char *metadata, const uint8_t *data, uint32_t length)
+{
+	/* prevent adding CRLF, which may crash spinel decoding */
+	uint32_t flag = LOG_OUTPUT_FLAG_CRLF_NONE;
+
+	log_backend_std_sync_hexdump(&log_output, flag, src_level,
+				     timestamp, metadata, data, length);
+}
+
+static void log_backend_spinel_init(void)
+{
+	memset(char_buf, '\0', sizeof(char_buf));
+}
+
+static void panic(struct log_backend const *const backend)
+{
+	log_backend_std_panic(&log_output);
+}
+
+static void dropped(const struct log_backend *const backend, uint32_t cnt)
+{
+	ARG_UNUSED(backend);
+
+	log_backend_std_dropped(&log_output, cnt);
+}
+
+static int write(uint8_t *data, size_t length, void *ctx)
+{
+	/* assuming SPINEL backend logs as platform region */
+	const int log_region_platform = 12;
+
+	otPlatLog(CONFIG_LOG_BACKEND_SPINEL_LEVEL, log_region_platform,
+	"%s", data);
```

Comment:
please improve indentation

Reply:
Done. Can you tell me where to find some indentation rules? As I can see in other files done by Nordic, when breaking line of method arguments, number of tabs is quite different. Sometimes its 1 tab, another 3 or 4.

Reply:
https://docs.zephyrproject.org/latest/contribute/index.html#coding-style
https://www.kernel.org/doc/html/latest/process/coding-style.html

Reply:
Not sure it is written there, but when you break line on function params you just align to the first parameter unless it is not possible